src/oscar/apps/es_search/facet.py

```diff
@@ -0,0 +1,185 @@
+        return str(bucket['key'])
+
+    def get_display_name(self, bucket):
+        return self.get_field_value(bucket)
+
+    def get_select_url(self, field_value):
+        """
+        :param field_value: Value of the bucket field
+        :return: A url that adds a filter for this value
+        """
+        url = self.base_url.append_query_param(
+            'selected_facets', '%s:%s' % (self.field_name, field_value))
+        return self.strip_pagination(url)
+
+    def get_deselect_url(self, field_value):
+        """
+        :param field_value: Value of the bucket field
+        :return: A url that removes the filter for this value
+        """
+        url = self.base_url.remove_query_param(
+            'selected_facets', '%s:%s' % (self.field_name, field_value))
+        return self.strip_pagination(url)
+
+    @staticmethod
+    def strip_pagination(url):
+        if url.has_query_param('page'):
+            url = url.remove_query_param('page')
+        return url.as_string()
+
+    def build_facet_result(self, bucket):
+        field_value = self.get_field_value(bucket)
+        facet = {
+            'name': self.get_display_name(bucket),
+            'count': bucket['doc_count'],
+            'show_count': True,
+            'selected': False,
+            'disabled': (int(bucket['doc_count']) == 0)
+        }
+        if field_value in self.selected_facets.get(self.field_name, []):
+            # This filter is selected - build the 'deselect' URL
+            facet['selected'] = True
+            facet['deselect_url'] = self.get_deselect_url(field_value)
+        else:
+            facet['select_url'] = self.get_select_url(field_value)
+
+        return facet
+
+    def get_facet_display_name(self):
+        """
+        :return: The title used when displaying this facet
+        """
+        label = self.field_name.capitalize()
+        if self.user_defined_facets.get(self.field_name, {}).get('label'):
+            label = self.user_defined_facets[self.field_name]['label']
+
+        return label
+
+    def get_facet(self):
+        return {
+            'name': self.get_facet_display_name(),
+            'results': [self.build_facet_result(bucket) for bucket in self.buckets]
+        }
+
+
+class RangeFacet(Facet):
+
+    """
+    NOTE: Elasticsearch's range aggregations are bit awkward because
+    they exclude the "to" value from the range. There is jiggery pokery here
+    to alter the display value of the upper bound.
+
+    Currently this code assumes integer ranges only.
+    """
+
+    @staticmethod
+    def get_field_value(bucket):
+        from_value = bucket.get('from', '')
+        to_value = bucket.get('to', '')
+        return '{}-{}'.format(from_value, to_value)
+
+    @staticmethod
+    def get_display_name(bucket):
+        from_value = int(bucket.get('from', 0))
+        to_value = int(bucket.get('to', 0))
+        if not from_value:
+            return _('Up to {}'.format(to_value))
```

Comment:
The way this is translated is wrong - the translated string will change dynamically based on `to_value`. Should be:

```
_('Up to %(to_value)s') % {'to_value': to_value}
```

Same for all the other translations in this method.